# **ONLINE RETAIL TRANSACTION ANALYSIS**

This dataset contains information on transactions made by customers through an online retail platform. The dataset includes data on the products that were purchased, the quantity of each product, the date and time of each transaction, the price of each product, the unique identifier for each customer who made a purchase, and the country where each customer is located. This dataset can be used to analyze customer behavior and preferences, identify popular products, and optimize pricing and marketing strategies. The dataset is well-suited for data analysis and machine learning applications, as it contains a large volume of transactional data that can be used to train predictive models and make data-driven decisions.

## Objectives

* Obtain the dataset from Kaggle
* Understand customer behaviour 
* Identify popular products 
* Identify customer purchasing habits
* Analyse seasonality to identify trends 
* Optimise pricing strategies 
* Identify customer geographic location 


## Inputs

* Dataset obtained from Kaggle - https://www.kaggle.com/datasets/abhishekrp1517/online-retail-transactions-dataset

## Outputs

* Clean dataset
* Generated visualisation for insights inline with the objectives

## Additional Comments

* I would like to propose various insights and recommendations to understand the objectives and how sales can improve




---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\moby1\\Documents\\Hackathon1_project\\Hackathon1\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\moby1\\Documents\\Hackathon1_project\\Hackathon1'

# Section 1:
## loading raw dataset and cleaning

It is crucial in the online retail industry who your customers are, the purchasing habits and what can be done in order to improve sales. Identifying returning customers allows a business strategy of promotions to increase and retain business from those customers. It is also important to understand what are the most popular products as well as products are not doing so good in sales. This in turn will aid inventory management as well as identifying discarding or bringing in new lines of products based on customers purchasing history.

I will now import the libraries needed for this project below.

In [23]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style('whitegrid')
import plotly.express as px
from sklearn.pipeline import Pipeline

In [29]:
import pandas as pd

df = pd.read_csv("C:\\Users\\moby1\\Documents\\Hackathon1_project\\Hackathon1\\Dataset\\raw_data\\Online_Retail.csv")

df.head()

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

           InvoiceDate  UnitPrice  CustomerID         Country  
0  2010-12-01 08:26:00       2.55       17850  United Kingdom  
1  2010-12-01 08:26:00       3.39       17850  United Kingdom  
2  2010-12-01 08:26:00       2.75       17850  United Kingdom  
3  2010-12-01 08:26:00       3.39       17850  United Kingdom  
4  2010-12-01 08:26:00       3.39       17850  United Kingdom

* The dataset has already been viewed in Microsoft Excel for understanding and what needs cleaning. Identifying that duplicates exist and within the dataset, missing cells in the description column.

In [37]:
df.isnull().sum()


InvoiceNo         0
StockCode         0
Description    1454
Quantity          0
InvoiceDate       0
UnitPrice         0
CustomerID        0
Country           0
dtype: int64

* Above we can see that missing cells exist within the dataset under the column description
* We will check now for duplicate rows

In [39]:
df.duplicated().sum()

5268

* Above we can see that duplicate rows exist and a count of 5268 duplicates which will need removing.

---

# Section 2

Section 2 content

---

NOTE

* You may add as many sections as you want, as long as it supports your project workflow.
* All notebook's cells should be run top-down (you can't create a dynamic wherein a given point you need to go back to a previous cell to execute some task, like go back to a previous cell and refresh a variable content)

---

# Push files to Repo

* In cases where you don't need to push files to Repo, you may replace this section with "Conclusions and Next Steps" and state your conclusions and next steps.

In [ ]:
import os
try:
  # create your folder here
  # os.makedirs(name='')
except Exception as e:
  print(e)
